File: FastMines_android/src/main/java/fmg/android/app/model/AppData.java
package fmg.android.app.model;

import fmg.core.app.model.MosaicBackupData;
import fmg.core.app.model.MosaicInitData;
import fmg.android.app.presentation.MenuSettings;
import fmg.core.types.EOpen;

/** Data model of the application to save/restore */
public final class AppData {

    private MosaicInitData mosaicInitData;
    private MosaicActivityBackupData mosaicActivityBackupData;
    private boolean splitPaneOpen;

    public AppData() {
        setDefaults();
    }

    public void setDefaults() {
        mosaicInitData = new MosaicInitData();
        mosaicActivityBackupData = null;
        splitPaneOpen = MenuSettings.DEFAULT_SPLIT_PANE_OPEN;
    }

    public MosaicInitData getMosaicInitData() {
        if (mosaicActivityBackupData != null) {
            MosaicInitData res = new MosaicInitData();
            MosaicBackupData mosaicBackupData = mosaicActivityBackupData.mosaicBackupData;
            res.setMosaicType(mosaicBackupData.mosaicType);
            res.setSizeField(mosaicBackupData.sizeField);
            res.setCountMines((int)mosaicBackupData.cellStates
                                                   .stream()
                                                   .filter(c -> c.getOpen() == EOpen._Mine)
                                                   .count());
            return res;
        }
        return mosaicInitData;
    }

    public void setMosaicInitData(MosaicInitData mosaicInitData) {
        this.mosaicInitData = mosaicInitData;
        if (mosaicInitData != null)
            this.mosaicActivityBackupData = null;
    }

    public MosaicActivityBackupData getMosaicActivityBackupData() {
        return mosaicActivityBackupData;
    }

    public void setMosaicActivityBackupData(MosaicActivityBackupData mosaicActivityBackupData) {
        this.mosaicActivityBackupData = mosaicActivityBackupData;
        if (mosaicActivityBackupData != null)
            this.mosaicInitData = null;
    }

    public boolean isSplitPaneOpen() {
        return splitPaneOpen;
    }

    public void setSplitPaneOpen(boolean splitPaneOpen) {
        this.splitPaneOpen = splitPaneOpen;
    }

}

File: FastMines_uwp/src/fmg/uwp/app/Model/AppData.cs
using System.Linq;
using Fmg.Core.App.Model;
using Fmg.Uwp.App.Presentation;

namespace Fmg.Uwp.App.Model {

    /// <summary> Data model of the application to save/restore </summary>
    public class AppData {

        private MosaicInitData mosaicInitData;
        private MosaicPageBackupData mosaicPageBackupData;
        public bool SplitPaneOpen { get; set; }

        public AppData() {
            SetDefaults();
        }

        public void SetDefaults() {
            mosaicInitData = new MosaicInitData();
            mosaicPageBackupData = null;
            SplitPaneOpen = MenuSettings.DEFAULT_SPLIT_PANE_OPEN;
        }

        public MosaicInitData MosaicInitData {
            get {
                if (mosaicPageBackupData != null) {
                    MosaicBackupData mosaicBackupData = mosaicPageBackupData.MosaicBackupData;
                    return new MosaicInitData {
                        MosaicType = mosaicBackupData.MosaicType,
                        SizeField = mosaicBackupData.SizeField,
                        CountMines = mosaicBackupData.CellStates
                                                     .Where(c => c.Open == Core.Types.EOpen._Mine)
                                                     .Count()
                    };
                }
                return mosaicInitData;
            }
            set {
                mosaicInitData = value;
                if (mosaicInitData != null)
                    mosaicPageBackupData = null;
            }
        }

        public MosaicPageBackupData MosaicPageBackupData {
            get {
                return mosaicPageBackupData;
            }
            set {
                mosaicPageBackupData = value;
                if (mosaicPageBackupData != null)
                    mosaicInitData = null;
            }
        }

    }

}
